src/hyperion/device_setup_plans/read_hardware_for_setup.py

```diff
@@ -19,24 +20,26 @@ def read_hardware_for_ispyb_pre_collection(
     undulator: Undulator,
     synchrotron: Synchrotron,
     s4_slit_gaps: S4SlitGaps,
+    robot: BartRobot,
 ):
-    LOGGER.info("Reading status of beamline parameters for ispyb deposition.")
+    LOGGER.info("Reading status of beamline for ispyb deposition, pre colection.")
     yield from bps.create(
         name=ISPYB_HARDWARE_READ_PLAN
     )  # gives name to event *descriptor* document
     yield from bps.read(undulator.current_gap)
     yield from bps.read(synchrotron.machine_status.synchrotron_mode)
     yield from bps.read(s4_slit_gaps.xgap)
     yield from bps.read(s4_slit_gaps.ygap)
+    yield from bps.read(robot.barcode)
```

Comment:
Is there anywhere that actually tests these methods? It's not clear to me that there is

Reply:
I think `test_read_hardware_for_ispyb_updates_from_ophyd_devices` tests them?